typing/typecore.ml

```diff
@@ -4197,11 +4192,11 @@ and type_expect_
       re {
         exp_desc = Texp_record {
             fields; representation;
-            extended_expression = opt_exp
+            extended_expression = opt_exp;
+            alloc_mode = Value_mode.regional_to_global_alloc expected_mode.mode
```

Comment:
It might be clearer to have the `alloc_mode` be optional here, since the mode for an unboxed record containing regional values will be misleading. That's because the unboxed record is being "allocated" as a regional value in that case -- which can't be expressed as an `Alloc_mode.t`.